src/code/z_message_PAL.c

```diff
@@ -1,66 +1,3285 @@
+                gDPSetEnvColor(gfx++, 10, 10, 10, 0);
+
+                gDPLoadTextureBlock(gfx++, sOcarinaNoteTextures[gOcarinaSongNotes[g].notesIdx[i]], G_IM_FMT_IA,
+                                    G_IM_SIZ_8b, 16, 16, 0, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP,
+                                    G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+
+                gSPTextureRectangle(gfx++, notePosX << 2, R_OCARINA_NOTES_YPOS(gOcarinaSongNotes[g].notesIdx[i]) << 2,
+                                    (notePosX + 16) << 2,
+                                    (R_OCARINA_NOTES_YPOS(gOcarinaSongNotes[g].notesIdx[i]) + 16) << 2, G_TX_RENDERTILE,
+                                    0, 0, 1 << 10, 1 << 10);
+            }
+        }
+
+        if (msgCtx->msgMode != MSGMODE_SCARECROW_LONG_RECORDING_START && msgCtx->msgMode != MSGMODE_MEMORY_GAME_START) {
+            for (notePosX = R_OCARINA_NOTES_XPOS, i = 0; i < 8; i++, notePosX += R_OCARINA_NOTES_XPOS_OFFSET) {
+                if (sOcarinaNoteBuf[i] == OCARINA_NOTE_INVALID) {
+                    break;
+                }
+
+                if (1) {}
+                if (sOcarinaNotesAlphaValues[i] != 255) {
+                    sOcarinaNotesAlphaValues[i] += VREG(50);
+                    if (sOcarinaNotesAlphaValues[i] >= 255) {
+                        sOcarinaNotesAlphaValues[i] = 255;
+                    }
+                }
+
+                gDPPipeSync(gfx++);
+                if (sOcarinaNoteBuf[i] == OCARINA_NOTE_A) {
+                    gDPSetPrimColor(gfx++, 0, 0, sOcarinaNoteAPrimR, sOcarinaNoteAPrimG, sOcarinaNoteAPrimB,
+                                    sOcarinaNotesAlphaValues[i]);
+                    gDPSetEnvColor(gfx++, sOcarinaNoteAEnvR, sOcarinaNoteAEnvG, sOcarinaNoteAEnvB, 0);
+                } else {
+                    gDPSetPrimColor(gfx++, 0, 0, sOcarinaNoteCPrimR, sOcarinaNoteCPrimG, sOcarinaNoteCPrimB,
+                                    sOcarinaNotesAlphaValues[i]);
+                    gDPSetEnvColor(gfx++, sOcarinaNoteCEnvR, sOcarinaNoteCEnvG, sOcarinaNoteCEnvB, 0);
+                }
+
+                gDPLoadTextureBlock(gfx++, sOcarinaNoteTextures[sOcarinaNoteBuf[i]], G_IM_FMT_IA, G_IM_SIZ_8b, 16, 16,
+                                    0, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK,
+                                    G_TX_NOLOD, G_TX_NOLOD);
+
+                gSPTextureRectangle(gfx++, notePosX << 2, R_OCARINA_NOTES_YPOS(sOcarinaNoteBuf[i]) << 2,
+                                    (notePosX + 16) << 2, (R_OCARINA_NOTES_YPOS(sOcarinaNoteBuf[i]) + 16) << 2,
+                                    G_TX_RENDERTILE, 0, 0, 1 << 10, 1 << 10);
+            }
+        }
+    }
+
+end:
+    *p = gfx;
+}
+
+void Message_DrawSetup(s16* scarecrowCustomSongSet, GraphicsContext* gfxCtx) {
+    static s16 sScarecrowCustomSongSet = 0;
+    static s16 sFillTimer = 0;
+    s32 pad;
```

Comment:
This function seems to be very specific for drawing something related to the scarecrow song. I think this function name `Message_DrawSetup` is a little too general. Do you know what this is drawing/setting up specifically?